### PR TITLE
GEOT-5913: WFS and WMTS: Abstract can not be read

### PR DESCRIPTION
**Pull request by matthias24**

### CODE REVIEW

**Comment by aaime** [comment]
Thanks @matthias24 for the effort, much appreciated, the fist pull request is always hard to setup.

Now... in order to be able accept it, there are a few other things needed (as listed in https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md), in particular:
* Contribution agreement (otherwise using this code may cause legal trouble for us in the future)
* JUnit tests

File: modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/v2_0/Capabilities200ServiceInfo.java [hunk 1]
```diff
@@ -142,7 +157,9 @@ public String getTitle() {
                 || serviceIdentification.getTitle().isEmpty()) {
             return null;
         }
-        return String.valueOf(serviceIdentification.getTitle().get(0));
+
+        return ((LanguageStringType) serviceIdentification.getTitle().get(0)).getValue();
+        // return String.valueOf(serviceIdentification.getTitle().get(0));
```
Comment (ianturton) [inline modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/v2_0/Capabilities200ServiceInfo.java]: you can just delete this line

File: modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/v2_0/Capabilities200ServiceInfo.java [hunk 2]
```diff
@@ -62,9 +68,18 @@ public String getDescription() {
...
-        @SuppressWarnings("unchecked")
-        List<String> abs = serviceIdentification.getAbstract();
-        return abs == null || abs.isEmpty() ? null : abs.get(0);
+
+        // The Abstract is of Type LanguageStringType, not String.
+        StringBuilder sb = new StringBuilder();
+        for (Object line : serviceIdentification.getAbstract()) {
+            if (line instanceof LanguageStringType) {
+                sb.append(((LanguageStringType) line).getValue());
+            } else {
```
Comment (ianturton) [inline modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/v2_0/Capabilities200ServiceInfo.java]: is line likely to be something other than a LanguageStringType?

File: modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/v2_0/Capabilities200ServiceInfo.java (hunk 1 above)
Comment (matthias24) [reply modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/v2_0/Capabilities200ServiceInfo.java]: Line 162? Yes I can. Should I?

File: modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/v2_0/Capabilities200ServiceInfo.java (hunk 2 above)
Comment (matthias24) [reply modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/v2_0/Capabilities200ServiceInfo.java]: Variable line is of type Object. So it is safer programming to allow it to be something different than LanguageStringType, too.

**Comment by matthias24** [comment]
What exactly failed? I saw the following in the logs:

[ERROR] The build will stop as there is local modifications
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-scm-plugin:1.9.5:check-local-modification (exec1) on project gt-wfs-ng: The build will stop as there is local modifications -> [Help 1]

**Comment by aaime** [comment]
@matthias24 you did not build locally and did not follow the coding conventions, the build finds it has to reformat and fails. Please run a local build, locally it will reformat without failing (on the build servers it is setup to fail to disallow commits that are not following the conventions).
The error message could indeed be better, I'll see what I can do to improve it (lack of time...)

**Comment by matthias24** [comment]
@aaime I was not able to build locally, because the format plugin of maven is missing in my eclipse config. I read, this plugin is not available in the eclipse market place. Actually I have not the time, to install and config maven separate and do the build. Sorry. If you would like to use my patch soon, maybe you could format this three files for me?
By the way: The contributor agreement is on the way.

**Comment by aaime** [comment]
@matthias24 I guess we can do the formatting for you, but you'll have to write the junit tests.

**Comment by matthias24** [comment]
@aaime Thank you. I wrote two test classes. The third was already there.

**Comment by aaime** [comment]
The travis build is failing in the modules modified by this pull request, it seems regressions were introduced.

**Comment by matthias24** [comment]
@aaime From the log I think its still just the wrong formating. You said, you could do this for me. Or have you found a other problem?

**Comment by aaime** [comment]
@matthias24 these are not formatting failures (from the Travis build logs):

```
Tests run: 2, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 0.034 sec <<< FAILURE! - in org.geotools.data.wfs.internal.v2_0.Capabilities200ServiceInfoTest
testGetTitle(org.geotools.data.wfs.internal.v2_0.Capabilities200ServiceInfoTest)  Time elapsed: 0.033 sec  <<< FAILURE!
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertNotNull(Assert.java:621)
	at org.junit.Assert.assertNotNull(Assert.java:631)
	at org.geotools.data.wfs.internal.v2_0.Capabilities200ServiceInfoTest.testGetTitle(Capabilities200ServiceInfoTest.java:64)

testGetDescription(org.geotools.data.wfs.internal.v2_0.Capabilities200ServiceInfoTest)  Time elapsed: 0 sec  <<< FAILURE!
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertNotNull(Assert.java:621)
	at org.junit.Assert.assertNotNull(Assert.java:631)
	at org.geotools.data.wfs.internal.v2_0.Capabilities200ServiceInfoTest.testGetDescription(Capabilities200ServiceInfoTest.java:57)

Running org.geotools.data.wfs.internal.v2_0.FeatureTypeInfoImplTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.025 sec - in org.geotools.data.wfs.internal.v2_0.FeatureTypeInfoImplTest
Running org.geotools.data.wfs.internal.v2_0.storedquery.ParameterTypeFactoryTest
Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.536 sec - in org.geotools.data.wfs.internal.v2_0.storedquery.ParameterTypeFactoryTest

Results :

Failed tests: 
  Capabilities200ServiceInfoTest.testGetTitle:64 null
  Capabilities200ServiceInfoTest.testGetDescription:57 null

Tests run: 163, Failures: 2, Errors: 0, Skipped: 3
````

**Comment by bencaradocdavies** [comment]
@matthias24 have you had a chance to investigate these test failures?

**Comment by matthias24** [comment]
There seems to be a conflict. It says "Only those with write access to this repository can merge pull requests.".

**Comment by aaime** [comment]
The "conflict" is a GIT one, the pull request has not been merged because of test failures, and now code has been modified in the main repo that conflicts with your changes.
You should:

- Update your branch to the latest version of the GeoTools code and resolve conflicts
- Fix the failing tests

**Comment by tbarsballe** [comment]
The same tests are still failing as of the latest changes. These failures must be fixed before this can be merged

**Comment by matthias24** [comment]
@tbarsballe
I changed the failing tests. Now the branch seems ready.

**Comment by tbarsballe** [comment]
Based on discussion above, I've run the formatter against the branch - PR with formatting changes is here: https://github.com/geotools/geotools/pull/2032

I'm going to close this in favour of that one, and wait to verify that the CI builds all pass.

**Pull request closed without merging**